pkg/target/kusttarget.go

```diff
@@ -44,6 +45,7 @@ import (
 // KustTarget encapsulates the entirety of a kustomization build.
 type KustTarget struct {
 	kustomization *types.Kustomization
+	decoder       ifc.Decoder
```

Comment:
The `decoder` and `tcfg` is more like a global configuration. They are initialized only once at the beginning of the program. I'm thinking if we can put them in different types and call an initialization step before we have any kustomize target. We can also avoid passing the decoder or other interfaces many times through function arguments.

Reply:
This is true - but in the long term global vars create a worse problem than passing dependencies via function args.  We can reduce the need for the latter with better structure and/or a dependency injection framework.